core/java/iesi-core/src/main/java/io/metadew/iesi/script/action/http/HttpExecuteRequest.java

```diff
@@ -132,6 +115,93 @@ protected boolean executeAction() throws NoSuchAlgorithmException, IOException,
+                    );
+        } else {
+            log.warn(MessageFormat.format(getActionExecution().getAction().getType().concat(" does not accept {0} as type for expectedStatusCode"),
+                    dataType.getClass()));
+            return new HashMap<>();
+        }
+    }
+
+    private List<HttpHeader> combineHeaders(List<HttpHeader> defaultHeaders) {
+        Map<String, String> headers = convertParameters(getParameterResolvedValue(HEADERS_KEY));
+        List<HttpHeader> headersRemaining;
+
+        if (headers.isEmpty()) {
+            return defaultHeaders;
+        }
+        //Override Headers to existing Headers
+        defaultHeaders.forEach(header -> {
+            String name = header.getName();
+            if (headers.containsKey(name)) {
+                header.setValue(headers.remove(name));
+            }
+        });
+
+        headersRemaining = headers.entrySet().stream()
+                .map(headerEntry -> new HttpHeader(headerEntry.getKey(), headerEntry.getValue())).collect(Collectors.toList());
+
+        return Stream.of(defaultHeaders, headersRemaining).flatMap(Collection::stream).collect(Collectors.toList());
+    }
+
+    private List<HttpQueryParameter> combineQueryParameter(List<HttpQueryParameter> defaultQueryParameter) {
+        Map<String, String> queryParameters = convertParameters(getParameterResolvedValue(QUERY_PARAMETERS_KEY));
+        if (queryParameters.isEmpty()) {
+            return defaultQueryParameter;
+        }
+        //Override Headers to existing Headers
+        defaultQueryParameter.forEach(queryParameter -> {
+            String name = queryParameter.getName();
+            if (queryParameters.containsKey(name)) {
+                queryParameter.setValue(queryParameters.remove(name));
+            }
+        });
+
+        List<HttpQueryParameter> queryParametersRemaining = queryParameters.entrySet().stream()
+                .map(headerEntry -> new HttpQueryParameter(headerEntry.getKey(), headerEntry.getValue())).collect(Collectors.toList());
+
+
+        return Stream.of(defaultQueryParameter, queryParametersRemaining).flatMap(Collection::stream).collect(Collectors.toList());
+    }
+
+    private AbstractMap.SimpleEntry<String, String> buildParameter(String param) {
+        String[] keyValues;
+        if (!param.contains("=")) {
+            throw new KeyValuePairException(String.format("The parameter %s should contain key value pair separated by the equals character < key=value >.", param));
+        }
+
+        keyValues = param.split("=");
+        if (keyValues.length > 2) {
+            throw new KeyValuePairException(String.format("The parameter %s should contain one key value pair, please remove additional separator character.", param));
+        }
+
+        return new AbstractMap.SimpleEntry<>(keyValues[0], keyValues[1]);
+    }
```

Comment:
maybe it is better to create a specific one for buildHttpHeader and buildQueryParameter. The you can also return a HttpHeader and HttpQueryParameter directly